go/adbc/driver/internal/driverbase/connection.go

```diff
@@ -714,4 +768,52 @@ func ValueOrZero[T any](val *T) T {
 	return *val
 }
 
+func maybeAddTraceParent(ctx context.Context, cnxn adbc.OTelTracing, st adbc.OTelTracing) (context.Context, error) {
+	var traceParentStr string
+	if st != nil && st.GetTraceParent() != "" {
+		traceParentStr = st.GetTraceParent()
+	} else if cnxn != nil && cnxn.GetTraceParent() != "" {
+		traceParentStr = cnxn.GetTraceParent()
+	}
+	if traceParentStr != "" {
+		spanContext, err := propagateTraceParent(ctx, traceParentStr)
+		if err != nil {
+			return ctx, err
+		}
+		ctx = trace.ContextWithRemoteSpanContext(ctx, spanContext)
+	}
+	return ctx, nil
+}
+
+func propagateTraceParent(ctx context.Context, traceParentStr string) (trace.SpanContext, error) {
+	if strings.TrimSpace(traceParentStr) == "" {
+		return trace.SpanContext{}, fmt.Errorf("traceparent string is empty")
+	}
+
+	propagator := propagation.TraceContext{}
+	carrier := propagation.MapCarrier{"traceparent": traceParentStr}
+	extractedContext := propagator.Extract(ctx, carrier)
+
+	spanContext := trace.SpanContextFromContext(extractedContext)
+	if !spanContext.IsValid() {
+		return trace.SpanContext{}, fmt.Errorf("invalid traceparent string")
+	}
+	return spanContext, nil
+}
+
+func isValidateTraceParent(traceParent string) bool {
+	// Supports version-format 00
+	// see: https://www.w3.org/TR/trace-context/#trace-context-http-headers-format
+	const tpPattern = `^(?<version>[0]{2})-(?<traceId>[0-9a-f]{32})-(?<parentId>[0-9a-f]{16})-(?<traceFlags>[0-9a-f]{2})$`
+	tpRegExp := regexp.MustCompile(tpPattern)
```

Comment:
can we push this to `init` or otherwise so that we don't do the regexp Compile on every call?

Reply:
Moved to a global variable/constant. Thanks.